vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -372,14 +368,23 @@ import { isFocusable } from '@vaadin/grid/src/vaadin-grid-active-item-mixin.js';
             page = Math.min(page, Math.floor(cache[parentUniqueKey].size / grid.pageSize));
 
             callback(cache[parentUniqueKey][page], cache[parentUniqueKey].size);
+            
+            // Flush after the callback to have the grid rows up-to-date
+            updateAllGridRowsInDomBasedOnCache();
+            // Prevent sub-caches from being created (& data requests sent) for items
+            // that may no longer be visible
+            ensureSubCacheQueue = [];
```

Comment:
During the rendering cycle, some expanded items that eventually end up outside the viewport may temporarily be inside the viewport. These lines are essential in making sure no unnecessary data requests get made once the hierarchy is completely rendered.